index.js

```diff
@@ -3983,6 +3983,26 @@
   S.splitOn =
   def('splitOn', {}, [$.String, $.String, $.Array($.String)], splitOn);
 
+  //# splitOnRegex :: RegExp -> String -> Array String
+  //.
+  //. Returns the substrings of its second argument separated by evaluated
+  //. matches of its first argument.
+  //.
+  //. See also [`splitOn`](#splitOn).
+  //.
+  //. ```javascript
+  //. > S.splitOnRegex(/\s+/, 'foo  bar baz   qux')
+  //. ['foo', 'bar', 'baz', 'qux']
+  //. ```
+  function splitOnRegex(pattern, s) {
+    return s.split(pattern);
+  }
+  S.splitOnRegex =
+  def('splitOnRegex',
+    {},
+    [$.RegExp, $.String, $.Array($.String)],
+    splitOnRegex);
```

Comment:
Parametric polymorphism (e.g. `Semigroup a => a -> a -> a`) is wonderful, but we avoid ad hoc polymorphism (e.g. `(RegExp | String) -> String -> Array String`) as it's [simpler][1] to have two functions which each do one thing than to have one function which does two things.


[1]: https://www.infoq.com/presentations/Simple-Made-Easy